### PR TITLE
Support local LLMs such as CodeLLaMa (llama-2 based)

### PR DESCRIPTION
**Pull request by aaronkaplan**
Hi!

I contributed a tiny set of patches so that gepetto can also access locally running LLMs such as [CodeLLama](https://huggingface.co/TheBloke/CodeLlama-34B-Instruct-GGUF) (or see [here](https://github.com/facebookresearch/codellama)). 

- Added documentation to the README.md file
- Added sample config in config.ini
- Added code to access it.

Ping me pls. in case you want to test it, I can demo it if needed.

Have a good week!

### CODE REVIEW

**Comment by JusticeRage** [comment]
Hi! Thanks a lot for this PR! Let me try to run codellama-2 locally if it's not too complicated, to see if I can test the code and make it work!

**Comment by aaronkaplan** [comment]
You are welcome! It takes a bit of resources (I am running it on an A100 GPU!) . You could try with [oobabooga](https://github.com/oobabooga/text-generation-webui) in llama.cpp mode (read: on CPUs / mixed GPU+CPU) in case you don't have the resources.
Or shoot me an email (see the link in the README.md) and we can set up a call so that I can demo it to you.

Note that sometimes codellama does not return proper JSON yet. That's something which probably could be addressed with things like langchain's output parsers.

File: gepetto/config.ini [hunk 1]
```diff
@@ -1,5 +1,7 @@
 [Gepetto]
```
Comment (captain-hoity-toity) [inline gepetto/config.ini:1]: Sure, here's a brief code review in markdown format:

---
## Code Review

### File 1: `filename.extension`

#### Line XX:

- Recommended Change: 

```diff
- Original code
+ Suggested code
```

#### Reasoning: 

Brief explanation of the suggested change.

---

### File 2: `filename.extension`

#### Line XX:

- Recommended Change: 

```diff
- Original code
+ Suggested code
```

#### Reasoning: 

Brief explanation of the suggested change.

---

Please apply these changes to improve the code quality. If you have any questions or need further clarification, feel free to ask.

---

Please replace the placeholders (`filename.extension`, `Line XX`, `Original code`, `Suggested code`, and `Brief explanation of the suggested change`) with the actual details from the PR commit.

**Comment by JusticeRage** [comment]
Hi! So, I've started playing with the code, to check if the UI works, etc. Seems to be good, but just trying to send a request to a non-existent local server, I was surprised to get a reply from... OpenAI! Is the `OPENAI_API_BASE` variable actually used?
Also, I see a few `# XXX` and `FIXME`, is this version ready to be merged or is it still a work in progress?

**Comment by aaronkaplan** [comment]
> Hi! So, I've started playing with the code, to check if the UI works, etc. Seems to be good, but just trying to send a request with the default config value (`http://3.78.xxx/v1`), I was surprised to get a reply from... OpenAI! Is that normal? Also, I see a few `# XXX` and `FIXME`, is this version ready to be merged or is it still a work in progress?

Let me check... it might be that you discovered a developer hack which was still in there.

**Comment by F474M0R64N4** [comment]
how to adapt to https://huggingface.co/Phind/Phind-CodeLlama-34B-v2

**Comment by F474M0R64N4** [comment]
Tested. It works! Thanks!

**Review by Billy1900 — APPROVED** [review]

**Comment by JusticeRage** [comment]
Hi! Since CodeLlama is now supported via Ollama, I think I will close this PR now. Any objections?

**Pull request closed without merging**